### PR TITLE
Support all other GitHub events

### LINKED ISSUE #145: GitHub Action fails with error "issues is not supported yet"
### Bug Description

We use [the following GitHub Action](https://github.com/canonical/checkbox/blob/main/.github/workflows/issues_to_jira.yaml) to create Jira cards when a new issue is raised in our GitHub project. 2 weeks ago, this action [started to fail](https://github.com/canonical/checkbox/actions/workflows/issues_to_jira.yaml) with the following output:

```
<13>Oct 27 01:26:02 ubuntu: GITHUB_EVENT_NAME: issues,   GITHUB_REPOSITORY: canonical/checkbox,   GITHUB_SHA: 710581418d25a5beaeb7ab85cef3bd9975028c4c
<11>Oct 27 01:26:02 ubuntu: issues is not supported yet. Please request it to be added on https://github.com/canonical/github-runner-operator/issues/new/choose
logger: file repo_check_output.txt: No such file or directory
<11>Oct 27 01:26:02 ubuntu: Stopping execution of jobs due to repository setup is not compliant with policies.
<11>Oct 27 01:26:02 ubuntu: This runner will be stopped or lost, please fix the setup of the repository, then rerun this job.
```

Since the error message states to request support for "issues" to be added here, here I am :)

### To Reproduce

1. Create an issue in the Checkbox project

This will trigger the "Issue to Jira" action.

Expected outcome: action passes and a Jira card is created.

Actual outcome: action fails and no Jira cards are created.

### Environment

N/A

### Relevant log output

```shell
N/A
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cbartz**
Applicable spec: n/a

### Overview

Support all other GitHub events by using the default repo policy compliance check endpoint, which is supported since https://github.com/canonical/repo-policy-compliance/pull/875 .

An end-to-end test for the `issues` event is added.

Closes #145 

### Rationale
Since https://github.com/canonical/repo-policy-compliance/pull/807 has landed, all events other than `pull_request` will only check collaborator permissions, so we could add support for the rest with just collaborator permissions.  

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Comment by cbartz** [comment]
Some integration tests are failing, which should be fixed when https://github.com/canonical/github-runner-operator/pull/156 lands.

**Review by jdkandersson — DISMISSED** [review]
Thanks!

File: templates/pre-job.j2 [hunk 1]
```diff
@@ -41,8 +41,17 @@ CURL_ARGS=(
...
 REPO_CHECK=1
 
+# Special Workflow dispatch repo-policy-compliance service check designed to fail:
+if [[ "${GITHUB_WORKFLOW}" == "Workflow Dispatch Failure Tests 2a34f8b1-41e4-4bcb-9bbf-7a74e6c482f7" ]]; then
+  logger -s "Running the test workflow for integration tests, this test is configured to fail"
+
+  curl "${CURL_ARGS[@]}" \
+      -X POST \
+      http://{{host_ip}}:8080/always-fail/check-run &> repo_check_output.txt
+  REPO_CHECK=$?
```
Comment (jdkandersson) [inline templates/pre-job.j2:51]: Weird that this shows up here?

Comment (cbartz) [reply templates/pre-job.j2:51]: I just merged main into this branch, which includes this code snippet.

**Review by arturo-seijas — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**